beacon_node/eth1/src/http.rs

```diff
@@ -407,19 +421,19 @@ pub async fn send_rpc_request(
         .map_err(|e| format!("Failed to receive body: {:?}", e))
 }
 
-/// Accepts an entire HTTP body (as a string) and returns the `result` field, as a serde `Value`.
-fn response_result(response: &str) -> Result<Option<Value>, String> {
+/// Accepts an entire HTTP body (as a string) and returns either the `result` field or the `error['message']` field, as a serde `Value`.
+fn response_result_or_error(response: &str) -> Result<Result<Value, Value>, String> {
```

Comment:
I'm not sure about this `Result<Result<Value, Value>, String>`, it feels like an anti-pattern and I think the present implementation will result in some less-than-ideal errors.

For example, if we get an `"error"` field on `fn get_deposit_logs_in_range` we'll log an error saying `"No result field was returned for deposit logs"`, when it would be ideal to indicate that we got an error and the value of it.

I'd suggest something like this:

```rust
#[derive(Debug)]
enum RpcError {
    NoResultField,
    Error(String)
}

fn response_result_or_error(response: &str) -> Result<Value, RpcError> {
    ...
}
```
And then in `fn get_deposit_logs_in_range` we can do:

```rust
    response_result_or_error(&response_body)
        .map_err(|e| format!("Unable to obtain deposit logs: {:?}", e)?
```

Reply:
Making an explicit error type would be better, agreed.

While it might be out of scope for this PR specifically, I feel like this gets back to [my comment](https://github.com/sigp/lighthouse/pull/2304#issuecomment-815539934) about supporting the entirety of the set of JSON-RPC 2.0 error conditions? After all, it seems like that's what Geth is sending to us anyway.

Reply:
> I feel like this gets back to my comment about supporting the entirety of the set of JSON-RPC 2.0 error conditions? After all, it seems like that's what Geth is sending to us anyway.

I don't feel so strongly either way about parsing the `error` struct instead of flattening it to a `String`. AFAIK, ultimately we just flatten it and present it to the user as a `String` anyway, so I'm not sure there's a whole lot to gain.

Reply:
> AFAIK, ultimately we just flatten it and present it to the user as a String anyway, so I'm not sure there's a whole lot to gain.

I think we err on the side of simplicity and just pass the string on, so I'm inclined to agree 
here.

I'll work on your proposed changes now.